components/homme/src/theta-l_kokkos/cxx/EquationOfState.hpp

```diff
@@ -250,7 +250,7 @@ class EquationOfState {
 
 public:
 
-  bool            m_theta_hydrostatic_mode;
+  int            m_theta_hydrostatic_mode;
```

Comment:
This local var can be made an int if bool causes problem on SYCL, but the `init` method should just accept a bool.